src/plots/linePlot.ts

```diff
@@ -425,6 +450,74 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let xAccessor = this.x().accessor;
+      let yAccessor = this.y().accessor;
+      let filteredIndices = this._filterDownsamplingSlope(dataset, indices, this.x().scale, this.y().scale, xAccessor, yAccessor);
+      return filteredIndices;
+    }
+
+    private _filterDownsamplingSlope(dataset: Dataset, indices: number[],
+      xScale: Scale<any, number>, yScale: Scale<any, number>, xAcessor: Accessor<any>, yAcessor: Accessor<any>) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+      let min: number;
+      let max: number;
+      let currentSlope: number;
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
+      filteredIndices.push(indices[0]);
+      function belongToCurBucket(i: number) {
+        let p1x = xScale.scale(xAcessor(data[indices[i]], indices[i], dataset));
```

Comment:
you made an assumption that xScale.scale exists which might not be the case
you can use `Plot._scaledAccessor` function instead